src/main/java/org/springframework/data/elasticsearch/core/ReactiveElasticsearchTemplate.java

```diff
@@ -135,6 +137,66 @@ public ReactiveElasticsearchTemplate(ReactiveElasticsearchClient client, Elastic
 		return save(entity, getIndexCoordinatesFor(entity.getClass()));
 	}
 
+	@Override
+	public <T> Flux<T> saveAll(Publisher<T> entities, IndexCoordinates index) {
+
+		Assert.notNull(entities, "Entities must not be null!");
+
+		return Flux.from(entities) //
+				.map(t -> operations.forEntity(t, converter.getConversionService())) //
+				.collectList() //
+				.flatMapMany(adaptibleEntities -> {
+					Iterator<AdaptibleEntity<T>> iterator = adaptibleEntities.iterator();
+					List<IndexQuery> indexRequests = adaptibleEntities.stream() //
+							.map(e -> getIndexQuery(e.getBean(), e)) //
+							.collect(Collectors.toList());
+					return doBulkOperation(indexRequests, BulkOptions.defaultOptions(), index) //
+							.map(bulkItemResponse -> {
+								AdaptibleEntity<T> mappedEntity = iterator.next();
+								mappedEntity.populateIdIfNecessary(bulkItemResponse.getResponse().getId());
+								return mappedEntity.getBean();
+							});
+				});
+	}
+
+	@Override
+	public <T> Flux<T> multiGet(Query query, Class<T> clazz, IndexCoordinates index) {
+
+		Assert.notNull(index, "index must not be null");
+		Assert.notEmpty(query.getIds(), "No Id define for Query");
+
+		MultiGetRequest request = requestFactory.multiGetRequest(query, index);
+		return Flux.from(execute(cl -> cl.multiGet(request))) //
+				.handle((result, sink) -> {
+
+					Document document = DocumentAdapters.from(result);
```

Comment:
Can a result be null? (i.e. do we regularly need to consider whether `source` is attached)?